### PR TITLE
Add `suggest` option to `DataCatalog._get_dataset`

### PR DESCRIPTION
**Pull request by deepyaman**
Signed-off-by: Deepyaman Datta <deepyaman.datta@utexas.edu>

## Description

See https://github.com/kedro-org/kedro-viz/pull/832

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1482"></a>

### CODE REVIEW

File: kedro/io/data_catalog.py [hunk 1]
```diff
@@ -282,15 +282,16 @@ class to be loaded is specified with the key ``type`` and their
...
     ) -> AbstractDataSet:
         if data_set_name not in self._data_sets:
             error_msg = f"DataSet '{data_set_name}' not found in the catalog"
 
-            matches = difflib.get_close_matches(data_set_name, self._data_sets.keys())
-            if matches:
-                suggestions = ", ".join(matches)  # type: ignore
-                error_msg += f" - did you mean one of these instead: {suggestions}"
+            if suggest:
+                matches = difflib.get_close_matches(data_set_name, self._data_sets.keys())
```
Comment (deepyaman) [inline kedro/io/data_catalog.py:291]:
```suggestion
                matches = difflib.get_close_matches(
                    data_set_name, self._data_sets.keys()
                )
```

**Comment by AntonyMilneQB** [comment]
🤔  What is this PR about?

> ## Description
> See #1482

image

**Comment by deepyaman** [comment]
> 🤔 What is this PR about?
> 
> > ## Description
> > See #1482
> 
> image

/facepalm

See https://github.com/kedro-org/kedro-viz/pull/832

**Comment by AntonyMilneQB** [comment]
Ah, I see, thank you! This looks good but we need to make sure it's part of 0.18.1 or kedro-viz won't actually work 😬

**Review by AntonyMilneQB — APPROVED** [review]

File: kedro/io/data_catalog.py [hunk 2]
```diff
@@ -282,15 +282,18 @@ class to be loaded is specified with the key ``type`` and their
...
+        self, data_set_name: str, version: Version = None, suggest: bool = True
     ) -> AbstractDataSet:
         if data_set_name not in self._data_sets:
             error_msg = f"DataSet '{data_set_name}' not found in the catalog"
 
-            matches = difflib.get_close_matches(data_set_name, self._data_sets.keys())
-            if matches:
-                suggestions = ", ".join(matches)  # type: ignore
-                error_msg += f" - did you mean one of these instead: {suggestions}"
+            if suggest:
```
Comment (MerelTheisenQB) [inline kedro/io/data_catalog.py:290]: Maybe add a comment to explain what this is for?

**Review by MerelTheisenQB — APPROVED** [review]
Nice one! 👍

File: kedro/io/data_catalog.py (hunk 2 above)
Comment (MerelTheisenQB) [inline kedro/io/data_catalog.py:290]:
```suggestion
           # Flag to turn on/off fuzzy-matching which can be time consuming and slow down plugins like `kedro-viz`
            if suggest:
```

**Pull request merged**